public/app/plugins/datasource/influxdb/datasource.test.ts

```diff
@@ -378,7 +381,14 @@ describe('InfluxDataSource Frontend Mode', () => {
       });
 
       it('should render chained regex variables with URL', () => {
-        ds.metricFindQuery('SHOW TAG VALUES WITH KEY = "agent_url" WHERE agent_url =~ /^$var1$/', {
+        const variablesMock = [queryBuilder().withId('var1').withName('var1').withCurrent('var1').build()];
+        const mockTemplateService = new TemplateSrv({
+          getVariables: () => variablesMock,
+          getVariableWithName: (name: string) => variablesMock.filter((v) => v.name === name)[0],
+          getFilteredVariables: jest.fn(),
+        });
+        let myDs = getMockInfluxDS(getMockDSInstanceSettings(), mockTemplateService);
```

Comment:
I wish I figured this way earlier. This is just the neat way of mocking `templateSrv`. In a separate PR, I'll update all weird `templateSrv` mocks.